src/bodywork/k8s/deployments.py

```diff
@@ -333,6 +334,15 @@ def monitor_deployments_to_completion(
     :return: True if all of the deployments are successful.
     """
     sleep(wait_before_start_seconds)
+
+    unschedulable_pods = [
+        has_unscheduleable_pods(deployment) for deployment in deployments
+    ]
+    if any(unschedulable_pods):
+        raise BodyworkClusterResourcesError(
+            "deployment", [deployment.metadata.name for deployment in deployments]
+        )
+
```

Comment:
Could also add this as a method to `utils.py` (e.g. check_cluster_resources) because it's in batch_jobs.py as well. It's only a tiny bit of repeated code but it will make it more readable as the formatting splits it on multiple lines.

Reply:
I think I didn't do this, because of the need to pass the resource type to the exception. I'll have another think about this, because I agree with the suggestion.

Reply:
I've sorted this out - much better 👍